app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments, class_name: 'Comment', foreign_key: 'post_id'
+  has_many :likes, class_name: 'Like', foreign_key: 'post_id'
+
+  def update_counter
+    author.increment(posts_counter)
+  end
```

Comment:
- [ ] [OPTIONAL] Your code would be better if you would use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `# update_counter` method every time a new post has been saved to the database, then make the `# update_counter` method private
**HINT:**
```ruby
after_save :update_post_counter

private

def update_counter
  author.increment!(:posts_counter)
end
```